src/devtools/views/Elements/TreeContext.js

```diff
@@ -307,17 +306,14 @@ function reduceSearchState(store: Store, state: State, action: Action): State {
     }
   }
 
-  // Changes in search index should override the selected element.
-  if (searchIndex !== prevSearchIndex) {
```

Comment:
The idea behind this was to avoid having to pay the cost of `getIndexOfElementID` every time the context re-rendered. That lookup isn't _too_ expensive but it's also something I'd like to avoid doing unnecessarily. In theory, if the index didn't change, we shouldn't need to look things up again.

I wonder if we could treat 0 as a special case...

Reply:
Did you come up with a small repro steps by chance? 😄

Reply:
Yeah, we could handle index 0 as a special case and updating the original if condition with an additional check for index 0 makes it work. Does this make sense?

Reproduction is quite easy (if I understand it correctly :wink:):
1. Go to https://react-devtools-experimental.now.sh/
2. In the devtools search box, type `list` - first element `<List>` should be selected
3. Continue typing and enter `listitem` - ideally the second element in the tree `<Memo(ListItem)>` should be selected but the first element is still shown as selected.


![react-devtools-repro](https://user-images.githubusercontent.com/450559/55325065-9a7f1b80-54a1-11e9-98df-33564d8e448c.gif)

You could see another example in my original screen grabs as well in the PR.